client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -52,22 +61,49 @@ const FormComponent = ({
   );
 };
 
-export const useJsonForms = (docName: string) => {
-  const [data, setData] = useState<JsonData>(patient);
-  const [loading, setLoading] = useState(false);
-  const [error] = useState(null);
+interface JsonFormOptions {
+  showButtonPanel?: boolean;
+  onCancel?: () => void;
+}
+
+export const useJsonForms = (
+  docName: string | undefined,
+  options: JsonFormOptions = {}
+) => {
+  const [data, setData] = useState<JsonData>(patient); // Replace with DB query hook
+  const [loading] = useState(false); // Replace with DB query hook
+  const [saving, setSaving] = useState(false);
+  const [error, setError] = useState<string | false>(false);
+  const t = useTranslation('common');
+  const { success, error: errorNotification } = useNotification();
+  const navigate = useNavigate();
+
+  const { showButtonPanel = true, onCancel = () => navigate(-1) } = options;
```

Comment:
I don't like introducing `useNavigate` which is specific to react router. 
I'd prefer to pass in a cancel action, that way the environment which is using the jsonForms can dictate what happens here.

Reply:
Yeah, I've got an "onCancel" prop for defining the cancel action, this is just the default.

You think I should set the default to just `() => {}` ?

Reply:
no, that's all good. I just missed the default aspect